com.unity.netcode.gameobjects/Tests/Runtime/ObjectParenting/NetworkObjectParentingTests.cs

```diff
@@ -86,6 +98,18 @@ public IEnumerator Setup()
 
             // Start server and client NetworkManager instances
             Assert.That(MultiInstanceHelpers.Start(true, m_ServerNetworkManager, m_ClientNetworkManagers));
+            m_ServerNetworkManager.SceneManager.ScenesInBuild.Add(nameof(NetworkObjectParentingTests));
+            // Register our scene verification delegate handler so we don't load the unit test scene
+            m_ServerNetworkManager.SceneManager.VerifySceneBeforeLoading = VerifySceneBeforeLoading;
+            foreach (var entry in m_ClientNetworkManagers)
+            {
+                if (!entry.SceneManager.ScenesInBuild.Contains(nameof(NetworkObjectParentingTests)))
+                {
+                    entry.SceneManager.ScenesInBuild.Add(nameof(NetworkObjectParentingTests));
+                }
+                // Register our scene verification delegate handler so we don't load the unit test scene
+                entry.SceneManager.VerifySceneBeforeLoading = VerifySceneBeforeLoading;
```

Comment:
I'm not sure if `VerifySceneBeforeLoading` implementation is really needed and should be exposed/rolled-out to public like this.

What we are saying fundamentally is that "in tests, do not care about scenes that have InitTestScene prefix".
This could be wrapped into a super small, focused block (with some macros and maybe with some runtime checks in like 10 LOC) and wouldn't require yetanother brand-new "VerifySceneBeforeLoading" mini feature.

Just my 2c though, if you're happy with this direction, I'm fine with it too.

Reply:
This is the feature for removing the scene registration.
The [RFC Explains what this is used for](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/blob/rfc/remove-sceneregistration/text/0000-remove-sceneregistration.md).
However, the cliff notes version is that this is a way for users to "secure" which scenes are loaded since there is no longer scene registration (which its original intention was to make sure you only allowed specific scenes to load).

I am using this in the unit test to exclude the loading of the unit test scene during the MultiInstance test for clients only.

Does this make sense?  It is a feature that I decided to use in the unit test to handle not re-loading the unit test scene as opposed to checking for that in NetworkSceneManager.

Reply:
oh! okay, I see. I didn't realize we actually care about "scene loading security" but sure, this is fine :)